R/srcref_df.R

```diff
@@ -258,6 +258,15 @@ test_trace_mapping <- function(x) {
 #' @param r A `list_of_srcref` object
 #'
 match_containing_srcrefs <- function(l, r) {
+  # NOTE:
+  #   srcrefs are matched due to filenames, as full file paths are often
+  #   different. Trace srcrefs srcfiles will point to a temporary installation
+  #   when running covr, whereas package namespace srcrefs srcfiles will point
+  #   to the package install location.
+  #
+  # TODO:
+  #   more comprehensive path comparisons (perhaps pkgname/R/file.R)
+
```

Comment:
@maksymiuks - I identified this potential pitfall when digging into this. Now that srcrefs might originate from other packages, just comparing basenames might not be sufficient. I also saw cases where srcrefs originated from the `inst` directory, which might have duplicated basenames also. 

This is probably an exceedingly rare edge case, but something we may want to think about.

Reply:
I agree it seems to be a rare case although it is good that it has been identified and we are aware that it can happen. What about keeping this in mind (like this TODO) and monitoring whether it needs addressing. Perhaps the rarity of this potential bug may cause that we will never end up in a situation when this gonna need addressing.

Reply:
I think we can create some heuristics to capture some parent directories in the file path comparison. Then we could differentiate

```
/home/user/R/lib/package/R/examples.R
/home/user/R/lib/dependency/inst/examples.R
```

while still treating

```
/home/user/R/lib/package/R/examples.R
/tmp/r_1234/R_LIB_covr_temp/package/R/examples.R
```

as stemming from the same code. 

As a start, taking the last 3 file path elements (`package/R/example.R`) should capture most common R package structure without introducing too much complexity.